src/Lib.hs

```diff
@@ -78,7 +79,9 @@ sma candles = SMA (totalSum candles / numCandles candles)
 
 -- | Using the old window data computes the next exponential moving average values
 getNextWindow :: ExchangeConf -> Window -> IO Window
-getNextWindow config oldWindow@(Window (shortEMA, longEMA) _) = do
+getNextWindow config oldWindow@(Window (shortEMA, longEMA) _) =
+  {- Making request to API will throw exception if over rate limit, just try again -}
+  flip catchIOError (\_ -> putStrLn "Failed request" >> return oldWindow) $ do
```

Comment:
So you don't have the same problem later, you'd like to be able to print out the status code & error message. However, the danger here is you're squashing any error here the same way.

Also, this is very Java-y and essentially an early return statement. 😡 ☕️ 👼 λ